ffi/src/handle.rs

```diff
@@ -156,19 +157,90 @@ macro_rules! gen_ffi_impl {
         $vis unsafe extern "C" fn __drop(
             handle_id: $crate::handle::Id,
             handle_ptr: *mut core::ffi::c_void,
-        ) -> $crate::FfiResult {
-            gen_ffi_impl!(@catch_unwind {
+        ) -> iroha_ffi::FfiReturn {
+            iroha_ffi::ffi_fn!(@catch_unwind {
                 match handle_id {
                     $( <$other as $crate::Handle>::ID => {
                         let handle_ptr = handle_ptr.cast::<$other>();
                         let handle: $other = $crate::TryFromReprC::try_from_repr_c(handle_ptr, &mut ())?;
                     } )+
                     // TODO: Implement error handling (https://github.com/hyperledger/iroha/issues/2252)
-                    _ => return Err($crate::FfiResult::UnknownHandle),
+                    _ => return Err(iroha_ffi::FfiReturn::UnknownHandle),
                 }
 
                 Ok(())
             })
         }
     };
 }
+
+/// Generate FFI equivalent declaration of the requested trait method (e.g. Clone, Eq, Ord)
+#[macro_export]
+#[cfg(feature = "client")]
+macro_rules! ffi_fn {
```

Comment:
Can we do better in terms of naming. @Arjentix @s8sato @Erigara ?

Reply:
I'll take suggestions. If `client` feature is not active `ffi_fn` will produce `extern` function implementations, if it's active it'll produce declarations

Reply:
That's silently changing the behaviour, and can lead to subtle bugs. I would instead prefer if 
- no `client` => `define_ffi_fn` is defined and `declare_ffi_fn` is not. 
- `client` active => `declare_ffi_fn` and `define_ffi_fn` is not.

Reply:
I think I resolved all of the concerns here